packages/plugin-core/src/test/suite-integ/BacklinksTreeDataProvider.test.ts

```diff
@@ -97,6 +101,53 @@ suite("BacklinksTreeDataProvider", function () {
     });
   });
 
+  test("with enableUnrefLinks from cache", (done) => {
+    let noteWithTarget: NoteProps;
+
+    runLegacyMultiWorkspaceTest({
+      ctx,
+      preSetupHook: async ({ wsRoot, vaults }) => {
+        noteWithTarget = await NOTE_PRESETS_V4.NOTE_WITH_TARGET.create({
+          wsRoot,
+          vault: vaults[0],
+        });
+      },
+      onInit: async ({ wsRoot, vaults }) => {
+        TestConfigUtils.withConfig(
+          (config) => {
+            config.dev = {
+              enableUnrefLinks: true,
+            };
+            return config;
+          },
+          { wsRoot }
+        );
+        const isUnrefLinkEnabled = TestConfigUtils.getConfig({ wsRoot }).dev
+          ?.enableUnrefLinks;
+        expect(isUnrefLinkEnabled).toBeTruthy();
+
+        // currently unref links are not getting populated on the first engine init for some reason.
+        // FIX: once above problem is fixed, fix this test.
+        await NOTE_PRESETS_V4.NOTE_WITH_UNREF_TARGET.create({
```

Comment:
might be because of the race condition mentioned earlier as we're checking for unlinked references while notes are still being indexed